pkg/gardenadm/cmd/init/init.go

```diff
@@ -60,6 +62,13 @@ func run(ctx context.Context, opts *Options) error {
 		return fmt.Errorf("failed bootstrapping control plane: %w", err)
 	}
 
+	if err := b.FS.MkdirAll(filepath.Dir(cmd.ConfigDirLocation), os.ModeDir); err != nil {
+		return fmt.Errorf("failed config directory location file %s: %w", cmd.ConfigDirLocation, err)
+	}
+	if err := b.FS.WriteFile(cmd.ConfigDirLocation, []byte(opts.ConfigDir), 0644); err != nil {
```

Comment:
Should we omit world-readability? There might be infrastructure credential we might not want everybody to be able to read.
WDYT?

```suggestion
	if err := b.FS.WriteFile(cmd.ConfigDirLocation, []byte(opts.ConfigDir), 0640); err != nil {
```